helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -604,7 +604,47 @@ public CloudConfig getCloudConfig(String clusterName) {
       return null;
     }
 
-    return new CloudConfig.Builder(record).build();
+    return new CloudConfig(record);
+  }
+
+  /**
+   * Delete cloud config fields (not the whole config)
+   * @param clusterName
+   * @param cloudConfig
+   */
+  public void deleteCloudConfigPatrial(String clusterName, CloudConfig cloudConfig) {
+    if (!ZKUtil.isClusterSetup(clusterName, _zkClient)) {
+      throw new HelixException("fail to delete cloud config. cluster: " + clusterName + " is NOT setup.");
+    }
+
+    HelixConfigScope scope =
+        new HelixConfigScopeBuilder(ConfigScopeProperty.CLOUD).forCluster(clusterName).build();
+    remove(scope, cloudConfig.getRecord());
+  }
+
+  /**
+   * Update cloud config
+   * @param clusterName
+   * @param cloudConfig
+   */
+  public void updateCloudConfig(String clusterName, CloudConfig cloudConfig) {
```

Comment:
Based on the current design, if user needs to use Java API instead of REST, they need to call HelixAdmin for add and remove, call ConfigAccessor for get, update and delete fields. Is this separation intended? Seems not intuitive.

Reply:
Basically, completely add or remove of cloud config seems to be admin work because it won't be used frequently (mainly used for cluster creation). I also find other similar API calls such as removeConfig in the HelixAdmin. However, updating or modifying such config are lower lever operations which have been implemented in ConfigAccessor.